File: python/greedy/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        current, farthest = 0, 0
        finalanswer = 0
        for i in range(len(nums) - 1):
            farthest = max(farthest, nums[i] + i)
            if i == current:
                current = farthest
                finalanswer += 1

        return finalanswer

File: c++/greedy/jump-game-ii.cc
class Solution {
public:
    int jump(vector<int>& nums) {
        int farthest = 0;
        int next = 0;
        int jumps = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (i > farthest) {
                farthest = next;
                jumps += 1;
            }
            next = std::max(next, i + nums[i]);
        }
        return jumps;
    }
};
